src/test/java/org/openwebnet4j/test/MessageTest.java

```diff
@@ -25,17 +25,31 @@
 import org.openwebnet4j.message.Lighting;
 import org.openwebnet4j.message.MalformedFrameException;
 import org.openwebnet4j.message.OpenMessage;
+import org.openwebnet4j.message.Thermoregulation;
 import org.openwebnet4j.message.UnsupportedFrameException;
 import org.openwebnet4j.message.WhereZigBee;
 
 /**
  * Tests for {@link BaseOpenMessage} and subclasses.
  *
  * @author M. Valla - Initial contribution
+ * @contributor G. Cocchi
  */
 
 public class MessageTest {
 
+    @Test
+    public void testThermoRegulationCommand() {
+        Thermoregulation thermoregulationMsg;
+        try {
+            thermoregulationMsg = (Thermoregulation) BaseOpenMessage.parse("*#4*2##");
+            assertNotNull(thermoregulationMsg);
+            System.out.println(thermoregulationMsg.toStringVerbose());
+        } catch (FrameException e) {
+            Assertions.fail();
+        }
+    }
+
```

Comment:
I would add to the same test method a couple of lines to test temperature serialization/deserialization is correct, also to test that only correct values are accepted.